src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +544,73 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+static bool replace_transpose_with_reshape(shared_ptr<Node> n)
+{
+    bool rc = false;
+    auto transpose = as_type_ptr<op::Transpose>(n);
+
+    if (transpose)
+    {
+        auto perm = as_type_ptr<op::Constant>(n->get_argument(1));
+        auto perm_value = perm->get_vector<int64_t>();
+        auto data = n->get_argument(0);
+        auto data_shape = n->input_value(0).get_partial_shape().to_shape();
+
+        // Data shape should be >= 3
+        if (data_shape.size() < 3 || data_shape.size() > 4)
+            return false;
+
+        // Check if input shape contains a Dim of value 1
+        int index_of_one = 0; // index of dim 1 in data_shape
+        for (int i = 0; i < data_shape.size(); i++)
+        {
+            if (data_shape[i] == 1)
+            {
+                index_of_one = i;
+                break;
+            }
+        }
+
+        if (!index_of_one)
+            return false;
+
+        // Remove index of 1 from perm_value and check if the other
+        // dimensions haven't interchanged
+        vector<int64_t> perm_value_copy = perm_value;
+
+        for (int i = 0; i < perm_value_copy.size(); i++)
```

Comment:
Since multiple 1s should be supported this for loop must be rewritten because you iterate over the vector which you modify (remove element). I would recommend another approach. Let's iterate over all elements in the perm_value. If the corresponding shape element is not one then we add the current perm_value element to the output vector. In the end we will have an output vector with indices of dimensions not equal to 1.